addons/website/static/src/snippets/s_popup/options.js

```diff
@@ -83,22 +83,28 @@ options.registry.SnippetPopup = options.Class.extend({
      * @override
      */
     cleanForSave: function () {
+        this._dirtyPage();
         this.$target.removeClass("s_popup_overflow_page");
     },
+    /**
+     * @override
+     */
+    onRemove() {
+        this._dirtyPage();
+    },
 
     //--------------------------------------------------------------------------
     // Options
     //--------------------------------------------------------------------------
 
     /**
-     * Moves the snippet in footer to be common to all pages
-     * or inside wrap to be on one page only
+     * Saves the location that the popup should have before saving.
      *
      * @see this.selectClass for parameters
      */
     moveBlock: function (previewMode, widgetValue, params) {
-        const $container = $(widgetValue === 'moveToFooter' ? 'footer' : 'main');
-        this.$target.closest('.s_popup').prependTo($container.find('.oe_structure:o_editable').first());
+        this.$target[0].closest('.s_popup').dataset.popupParent = widgetValue === 'moveToFooter' ?
+            'footer' : 'main';
```

Comment:
Maybe. But in any case, you should not rely on the fact that `publicWidget` code is moving your *edited* DOM at the right location. At minimum, that makes the code hard to read (logic spread across multiple files and widgets) for no reason.